File: Fall 2016/Demo 10/Demo 10/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace Demo_10
{
    public enum EyeColour
    {
        Brown=1, Blue, Green=4, Hazel=8,
        Lavender =16, Black=32
    }

    public enum Gender
    {
        Female, Male
    }

    public struct Person
    {
        public String Name;
        public EyeColour Eyes;
        public Gender Gender;
        public System.Drawing.Color Hair;
        public uint Age;
        public double Mass;
        public double Height;

        public override string ToString()
        {
            return Name
                + "(" + Gender + ", " + Age + ")"
                + ": " + Hair + ", " + Eyes + " : "
                + Mass + "kg, " + Height + "cm";
        }
    }//No need for semi at end
}

File: Fall 2018/Demo 2/person.c
#include "person.h"
#include <stdio.h>

//Print a person in
void PrintPerson(Person p)
{
    printf("%s (%d)\n\tHair:", p.Name,p.ID);
    PrintColour(p.Hair);
    printf(" Eyes:");
    PrintColour(p.Eyes);
    printf("\n");
    return;
}

void PrintColour(int c)
{
    switch(c)
    {
    case brown:
        printf("Brown"); break;
    case blue:
        printf("Blue"); break;

    case green:
        printf("Green"); break;

    case hazel:
        printf("Hazel"); break;

    case purple:
        printf("Purple"); break;

    case pink:
        printf("Pink"); break;

    case grey:
        printf("Grey"); break;

    case brunette:
        printf("Brunette"); break;

    case blonde:
        printf("Blonde"); break;

    case black:
        printf("Black"); break;
    case white:
        printf("White"); break;
    case  red:
        printf("Red"); break;
    default:
        printf("None");

    }
}
